reflex/components/component.py

```diff
@@ -677,13 +787,13 @@ def _get_mount_lifecycle_hook(self) -> str | None:
         """
         # pop on_mount and on_unmount from event_triggers since these are handled by
         # hooks, not as actually props in the component
-        on_mount = self.event_triggers.pop(EventTriggers.ON_MOUNT, None)
-        on_unmount = self.event_triggers.pop(EventTriggers.ON_UNMOUNT, None)
-        if on_mount:
+        on_mount = self.event_triggers.get(EventTriggers.ON_MOUNT, None)
+        on_unmount = self.event_triggers.get(EventTriggers.ON_UNMOUNT, None)
+        if on_mount is not None:
             on_mount = format.format_event_chain(on_mount)
-        if on_unmount:
+        if on_unmount is not None:
             on_unmount = format.format_event_chain(on_unmount)
-        if on_mount or on_unmount:
+        if on_mount is not None or on_unmount is not None:
```

Comment:
we might not have set the values on the previous lines...

when on_mount or on_unmount is not None, then we format it into an event chain; but if neither one is set, then we don't format_event_chain at all.

we only want to format the actual useEffect if either of those are set, otherwise no useEffect is needed.